File: week1/TwoSum/twoSum.cpp
#include <iostream>
using namespace std;

vector<int> twoSum(vector<int>& numbers, int taregt);

int main(){
    vector<int> input = {2,7,11,15};
    int taregt = 9;
    vector<int> output = twoSum(input, taregt);
    for(auto it: output){
        cout << it << '\n';
    }
}

vector<int> twoSum(vector<int>& numbers, int taregt){
    vector<int> output = {};
    for(int i = 0; i < numbers.size();i++){
        for(int j=0; j < numbers.size(); j++){
            if(numbers[i]+numbers[j]==taregt){
                output.push_back(i);
                output.push_back(j);
                return output;
            }
        }
    }

    return output;
}

File: week1/TwoSum/twoSum.py
class Solution:
    @classmethod
    def twoSum(self, numbers: list[int], taregt: int)->list[int]:
        output = []
        n = len(numbers)
        for i in range (n):
            for j in range (i+1, n):
                if numbers[i]+numbers[j]==target:
                    output.append(i)
                    output.append(j)
                    return output

        return output

input = [2,7,11,15]
target = 9
printOut = Solution.twoSum(input, target)
print(printOut)